src/hooks/sdk/useMediaStreamMetrics.tsx

```diff
@@ -0,0 +1,93 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useState, useEffect } from 'react';
+import { ClientMetricReport } from 'amazon-chime-sdk-js';
+
+import { useAudioVideo } from '../../providers/AudioVideoProvider';
+
+function isValidMetric(metric: any) {
+  return typeof metric === 'number' && !Number.isNaN(metric);
+}
+
+interface MediaStreamMetrics {
+  audioPacketsSentFractionLossPercent: number | null; // Percentage of audio packets lost (1s) from client to server
+  audioPacketsReceivedFractionLossPercent: number | null; // Percentage of audio packets lost from server to client
+  availableOutgoingBandwidth: number | null;
+  availableIncomingBandwidth: number | null;
+  videoStreamMetrics: { [attendeeId: string]: { [ssrc: string]: {[key: string]: number} } }
+}
+
+export function useMediaStreamMetrics() {
+  const audioVideo = useAudioVideo();
+  const [mediaStreamMetrics, setMediaStreamMetrics] = useState<MediaStreamMetrics>({
+    audioPacketsSentFractionLossPercent: null,
+    audioPacketsReceivedFractionLossPercent: null,
+    availableOutgoingBandwidth: null,
+    availableIncomingBandwidth: null,
+    videoStreamMetrics: {},
+  });
+
+  useEffect(() => {
+    if (!audioVideo) {
+      return;
+    }
+
+    const observer = {
+      metricsDidReceive(clientMetricReport: ClientMetricReport): void {
+        const {audioPacketLossPercent, audioPacketsReceivedFractionLoss, availableSendBandwidth, availableReceiveBandwidth, availableOutgoingBitrate, availableIncomingBitrate } = clientMetricReport.getObservableMetrics();
```

Comment:
I think this needs to be still prettified / indented right?

Something like this :
```js
const {
  audioPacketLossPercent,
  audioPacketsReceivedFractionLoss,
  availableSendBandwidth,
  availableReceiveBandwidth,
  availableOutgoingBitrate,
  availableIncomingBitrate 
} = clientMetricReport.getObservableMetrics();
```

Reply:
yeah, somehow the prettier formatter does't automatically change it for me. I will double check that.